create-container.c

```diff
@@ -1247,6 +1271,215 @@ int main(int argc, char* argv[])
 			die(EX_SOFTWARE, "Cannot write container header (r = %d) (%s)", r,
 			    strerror(errno));
 
+	} else {
+		// VERSION 3 CONTAINER
+
+		c_v3->magic_number = cpu_to_be32(ROM_MAGIC_NUMBER);
+		c_v3->version = cpu_to_be16(3);
+		c_v3->container_size = cpu_to_be64(SECURE_BOOT_HEADERS_V2_SIZE + payload_st.st_size);
+		memset(c_v3->hw_pkey_a, 0, sizeof(ecc_key_t));
+		memset(c_v3->hw_pkey_d, 0, sizeof(dilithium_key_t));
+		if (params.hw_keyfn_a) {
+			getPublicKeyRaw(&pubkeyraw, params.hw_keyfn_a);
+			verbose_print((char *) "pubkey A = ", pubkeyraw, sizeof(pubkeyraw));
+			memcpy(c_v3->hw_pkey_a, pubkeyraw, sizeof(ecc_key_t));
+		}
+		if (params.hw_keyfn_d) {
+			size_t sLen = sizeof(c_v3->hw_pkey_d);
+			int r = readBinaryFile(c_v3->hw_pkey_d, &sLen,params.hw_keyfn_d);
+			if (0 != r || sLen != DILITHIUM_PUB_KEY_LENGTH)
```

Comment:
For now we can leave this code such that it will put the current dilithium key in for key D & S .  Once this code is merged this will then change to using the MLDSA-87 algorithm.  To that end please create a test/v3_keys directory and copy over the existing v2_keys.  That way a caller can create a v3 container and pass in the v3 keys (even though those keys are actually dilithium and not mldsa).  Then in the next commit we would actually generate real v3 imprint mldsa keys.